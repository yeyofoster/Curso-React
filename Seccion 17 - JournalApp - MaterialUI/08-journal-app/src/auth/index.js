export * from './layout';
export * from './pages';
export * from './routes';