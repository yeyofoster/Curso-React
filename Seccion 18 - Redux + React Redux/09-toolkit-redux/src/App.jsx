import { useDispatch, useSelector } from 'react-redux'
import { decrement, increment, incrementBy } from './store/slices/counter';
import reactLogo from './assets/react.svg'
import viteLogo from '/vite.svg'
import './App.css'

function App() {
  const { counter } = useSelector((state) => state.counter);
  const dispatch = useDispatch();

  return (
    <>
      <div>
        <a href="https://vitejs.dev" target="_blank">
          <img src={viteLogo} className="logo" alt="Vite logo" />
        </a>
        <a href="https://react.dev" target="_blank">
          <img src={reactLogo} className="logo react" alt="React logo" />
        </a>
      </div>
      <h1>Vite + React</h1>

      <h2>Counter: {counter}</h2>
      <div className="card">
        <button onClick={() => dispatch(increment())}>
          Increment
        </button>
      </div>

      <div className="card">
        <button onClick={() => dispatch(decrement())}>
          Decrement
        </button>
      </div>

      <div className="card">
        <button onClick={() => dispatch(incrementBy(5))}>
          Increment by 5
        </button>
      </div>
    </>
  )
}

export default App
