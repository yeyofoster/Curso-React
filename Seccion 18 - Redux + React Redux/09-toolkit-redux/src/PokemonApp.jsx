export const PokemonApp = () => {
    return (
        <>
            <h1>PokemonApp</h1>
            <hr />

            <ul>
                <li>Hola</li>
                <li>Pokemon</li>
                <li>App</li>
            </ul>
        </>
    )
}
