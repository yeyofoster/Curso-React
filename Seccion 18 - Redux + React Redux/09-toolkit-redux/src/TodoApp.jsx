export const TodoApp = () => {
    return (
        <>
            <h1>Todos - RTK Query</h1>
            <hr />
            <h4>isLoading...</h4>

            <pre>
                ...
            </pre>

            <button
                style={{ backgroundColor: 'coral' }}
                onClick={() => { }}
            >
                Next Todo
            </button>
        </>
    );
};