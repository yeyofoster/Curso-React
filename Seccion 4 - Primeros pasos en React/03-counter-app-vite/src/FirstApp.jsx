export const FirstApp = () => {
    return (
        <>
            <h1>First App</h1>
            <p>Lorem ipsum dolor sit amet consectetur adipisicing elit. Minima pariatur eos maxime at deleniti iure reiciendis sed? Delectus explicabo ad dicta voluptate excepturi quidem repellat ea voluptas, amet porro sint!</p>
        </>
    )
};