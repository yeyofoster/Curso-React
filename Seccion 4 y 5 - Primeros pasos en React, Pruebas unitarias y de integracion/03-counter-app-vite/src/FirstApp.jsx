import PropTypes from 'prop-types';

export const FirstApp = ({ title, subTitle, name }) => {
    return (
        <>
            <h1>{title}</h1>
            <p>{subTitle}</p>
            <p>{name}</p>
        </>
    );
};

FirstApp.propTypes = {
    subTitle: PropTypes.number,
    title: PropTypes.string.isRequired
};

FirstApp.defaultProps = {
    name: 'Yeyo Foster',
    subTitle: 0
    // title: 'No hay titulo'
};