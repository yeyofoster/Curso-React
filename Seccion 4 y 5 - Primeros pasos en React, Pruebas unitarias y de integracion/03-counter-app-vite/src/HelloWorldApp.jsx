// function App() {
//     return (<h1>Hola mundo!!!</h1>);
// };

// export default App;

import React from 'react'

export const HelloWorldApp = () => {
    return (
        <h1>Hello World App</h1>
    )
};