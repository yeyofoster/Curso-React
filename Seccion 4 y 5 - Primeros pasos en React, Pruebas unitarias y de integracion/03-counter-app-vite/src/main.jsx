import React from 'react';
import ReactDOM from 'react-dom/client';
// import { HelloWorldApp } from './HelloWorldApp';
import { FirstApp } from './FirstApp';
// import { CounterApp } from './CounterApp';
import './styles.css';

ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp title='React'/>
    </React.StrictMode>
);