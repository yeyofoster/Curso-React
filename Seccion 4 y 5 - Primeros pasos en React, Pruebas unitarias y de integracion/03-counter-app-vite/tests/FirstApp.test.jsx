import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe('Pruebas en FirstApp', () => {

    test('Debe de hacer match con el snapshot', () => {
        const title = 'Hola soy Yeyo';
        const { container } = render(<FirstApp title={title} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el titulo en un h1', () => {
        const title = 'Hola soy Yeyo';
        const { container, getByText } = render(<FirstApp title={title} />);

        expect(getByText(title)).toBeTruthy();

        const h1 = container.querySelector('h1');

        // expect(h1.innerHTML).toBe(title);
        expect(h1.innerHTML).toContain(title);
    })
});