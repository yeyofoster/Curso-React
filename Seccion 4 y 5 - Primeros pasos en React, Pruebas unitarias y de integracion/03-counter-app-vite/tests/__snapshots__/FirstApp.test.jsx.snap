// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en FirstApp Debe de hacer match con el snapshot 1`] = `
<div>
  <h1>
    Hola soy Yeyo
  </h1>
  <p>
    0
  </p>
  <p>
    Yeyo Foster
  </p>
</div>
`;
